src/wrappers/cobs.py

```diff
@@ -1,15 +1,97 @@
+    def __init__(self):
+        index_paths = []
+        for path in Path(COBS_CLASSIC_INDEXES_DIR).iterdir():
+            if path.is_dir():
+                index_paths += [p for p in path.iterdir() if p.is_file()]
+        self.search_instances = [cobs.Search(str(index_path)) for index_path in index_paths]
 
     def search(self, query, threshold):
-        return self.search_instance.search(query, threshold=threshold)
+        results = []
+
+        for search_instance in self.search_instances:
+            results += search_instance.search(query, threshold=threshold)
+
+        flatten = []
+        for r in results:
+            flatten.append((r.score, r.doc_name))
+        return flatten
+
+    @staticmethod
+    def build(sample_paths, sample_names):
+        assert len(sample_paths) == len(sample_names)
+
+        p = cobs.ClassicIndexParameters()
+        p.term_size = COBS_TERM_SIZE
+        p.false_positive_rate = COBS_POSITIVE_RATE
+        setattr(p, 'continue', True)
+
+        Cobs.group_samples_by_signature_size(sample_paths, sample_names, p)
+        Cobs.build_new_indexes(p)
+        Cobs.combine_classic_indexes(p)
+
+        for sample_dir in Path(COBS_SAMPLE_DIR).iterdir():
+            for sample in sample_dir.glob('*'):
+                sample.unlink()
+
+    @staticmethod
+    def group_samples_by_signature_size(sample_paths, sample_names, params):
+        # cobs.DocumentEntry doesn't have a constructor, so the only way to construct one is to iterate a DocumentList
+        with TemporaryDirectory() as tmpdir:
+            doclist = cobs.DocumentList(tmpdir)
+            for original_sample_path in sample_paths:
+                doclist.add(original_sample_path.strip())
+
+            # Copy sample file to the group with the smallest signature size that is larger than its own
+            for doc, name in zip(doclist, sample_names):
+                signature_size = cobs.calc_signature_size(doc.num_terms(COBS_TERM_SIZE), params.num_hashes, params.false_positive_rate)
+                for index_dir in Path(COBS_SAMPLE_DIR).iterdir():
+                    if index_dir.is_dir() and int(index_dir.name) > signature_size:
+                        file_extension = doc.path.split('.')[-1]
+                        new_sample_path = index_dir / (name + '.' + file_extension)
+                        shutil.copyfile(doc.path, new_sample_path)
+                        break
+
+    @staticmethod
+    def build_new_indexes(params):
+        for sample_dir in Path(COBS_SAMPLE_DIR).iterdir():
+            doclist = cobs.DocumentList(str(sample_dir))
+            for doc in doclist:
+                doc.name = Path(doc.path).name.split('.')[-2]
+
+            classic_path = Path(COBS_CLASSIC_INDEXES_DIR) / sample_dir.name / (datetime.now().strftime("%Y-%m-%d_%H:%M:%S") + '.' + COBS_CLASSIC_FILE_EXTENSION)
+            cobs.classic_construct_list(doclist, str(classic_path), params)
```

Comment:
This will not build the classic index with the correct signature size. It will build with the signature size for the largest file in the input  doc list, which could be different from the `sample_dir.name` here.